### PR TITLE
Team Management Part 2

### LINKED ISSUE #1: Team Management
The platform needs a robust set of user and team management features, including:

### Structure

The 'main' HOLPA interface will have multi-tenancy. Most users will be team members, and the main interface will be at the team level. For users with a single team, all pages are filtered to show only things related to that team. For users with multiple teams, they can switch between teams, but only ever interact with 1 team at a time. 

There is also a 'program' level. A program is made of multiple teams, and intended for situations where there are multiple implementations of HOLPA that all fall under the same program or project. 

For example, the implementation of HOLPA in 8 countries by the core team can be considered 1 program.

A user can be a member of many teams;
A team can be a part of a single 'program'; 
A program is a set of implementations of HOLPA led by a specific organisation/project/program.  The idea is that program leads will have full access to all teams within the program. Team members would still only have access to their own team. 

### Site Admin Panel

- [ ] A User Resource page where admins can:
    - See the full list of registered users, their roles and teams;
    - Send a password reset link to a specific user;
    - Suspend or delete a user;
    - Invite a new user with a specific site-wide role;
    - Invite a new user with to a specific team or program (and with a specific team-wide or program-wide role);
 
- [ ] A Team Resource page where admins can:
    - See the full list of active teams, with space to show the 'status' of the team's progress with the HOLPA implementation. 
    - Create new teams
    - Edit team information
    - Delete teams (?) - need to prevent this when there is actual information / data collected.

- [ ] A Program Resource page where admins can:  
   - See the full list of registered program.
   - Create new programs, and create multiple teams for a program,
   - Edit program information
   - Delete programs (?) - need to prevent this when there is actual information / data collected.

### Program Admin Panel

- [ ] A team and user resource similar to the above.
- [ ] The ability to edit the current program information. 

### Team Panel

- [ ] A user resource similar to the above. 
- [ ] The ability to edit the current team information.

### PR DESCRIPTION
**Pull request by dan-tang-ssd**
This PR is submitted to fix #1 

It contains some features listed in #1. Considering the project milestones, some less important features will be developed in later stage.

---

This PR contains below changes:
1. Update seeder file, add description to the team record to remind user to create a new team for xlsform related testing. Because the team created by seeder file does not have a ODK project ID.
2. admin panel, Team resource and Program resource, add warning message when delete team or delete program, no matter if there is actual survey data collected
3. app panel, add Team resource for editing team details; add users, invites, xlsforms relation manager for read-only purpose

---

Less important features to be developed in later stage:
1. admin panel > User resource > Send a password reset link to a specific user
2. admin panel > User resource > Suspend or delete a user
3. program admin panel > use Program model for multi-tenancy
4. program admin panel > Users resource, only show teams under selected program in mutli-tenancy

### CODE REVIEW

**Comment by dan-tang-ssd** [comment]
This PR is now ready for review.

Most features are implemented for issue #1.

---

There are some minor features that we can implement later:

User resource:
 - Send a password reset link to a specific user;
    - we have "Forget password" feature in app panel login page
 - Suspend or delete a user;

Team resource:
 - Delete teams (?) - need to prevent this when there is actual information / data collected.
    - currently we show a warning message no matter this team has collected actual survey data or not

Program resource:
 - Delete programs (?) - need to prevent this when there is actual information / data collected.
    - currently we show a warning message no matter this team has collected actual survey data or not

**Comment by dan-tang-ssd** [comment]
Oh... forget to mention that this PR also fix #19

File: app/Filament/Admin/Resources/ProgramResource.php [hunk 1]
```diff
@@ -25,13 +25,16 @@ public static function form(Form $form): Form
...
-                Forms\Components\TextInput::make('description')
-                    ->maxLength(255),
-                Forms\Components\TextInput::make('note')
-                    ->maxLength(255),
+                Forms\Components\Section::make('Program Details')
+                    ->schema([
+                        Forms\Components\TextInput::make('name')
+                            ->required()
+                            ->maxLength(255),
+                        Forms\Components\TextInput::make('description')
```
Comment (dave-mills) [inline app/Filament/Admin/Resources/ProgramResource.php:33]: 255 characters is a little small for a description; I would typically go for a textarea entry and text field type for things like "description", "notes" etc.

File: app/Filament/Admin/Resources/ProgramResource/RelationManagers/TeamsRelationManager.php [hunk 2]
```diff
@@ -29,7 +42,26 @@ public function table(Table $table): Table
         return $table
             ->recordTitleAttribute('name')
             ->columns([
-                Tables\Columns\TextColumn::make('name'),
+                Tables\Columns\TextColumn::make('name')
+                    ->searchable()
+                    ->sortable(),
+                Tables\Columns\TextColumn::make('programs.name')
```
Comment (dave-mills) [inline app/Filament/Admin/Resources/ProgramResource/RelationManagers/TeamsRelationManager.php:48]: Not sure we need the program name in the relation manager here, as we know what program we're looking at from the page.

**Review by dave-mills — APPROVED** [review]
Looks good, works nicely for inviting users to programs or teams; Great as a base setup for the platform.

File: app/Filament/Admin/Resources/ProgramResource.php (hunk 1 above)
Comment (dan-tang-ssd) [reply app/Filament/Admin/Resources/ProgramResource.php:33]: As discussed, I will update this later.

**Comment by dan-tang-ssd** [comment]
By the way, I got a 30 seconds timeout issue when creating xlsform template for HOLPA Household ODK Form (159 KB).
Maybe it is a long ODK form so it takes more time to process, then lead to timeout.

I cannot create xlsform template for HOLPA Household ODK Form in local env successfully.
But I can do so in staging env without timeout issue.

**Comment by dave-mills** [comment]
> I cannot create xlsform template for HOLPA Household ODK Form in local env successfully.
> But I can do so in staging env without timeout issue.

The fact that the same form works on the staging env suggests that it's not due to the size of the form. Which means it's probably something about the local environment setup. The main thing for now is that the staging site is working for us to use the demo forms.

**Pull request merged**